shanoir-ng-users/src/main/java/org/shanoir/ng/events/EventsApi.java

```diff
@@ -0,0 +1,34 @@
+package org.shanoir.ng.events;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import org.shanoir.ng.shared.exception.RestServiceException;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import java.util.List;
+
+@Tag(name = "events", description = "the events API")
+@RequestMapping("/events")
+public interface EventsApi {
+
+    @Operation(summary = "", description = "If exists, returns the events corresponding to the given study id")
+    @ApiResponses(value = { @ApiResponse(responseCode = "200", description = "found user"),
+            @ApiResponse(responseCode = "401", description = "unauthorized"),
+            @ApiResponse(responseCode = "403", description = "forbidden"),
+            @ApiResponse(responseCode = "404", description = "no user found"),
+            @ApiResponse(responseCode = "500", description = "unexpected error") })
+    @RequestMapping(value = "/{studyId}", produces = { "application/json" }, method = RequestMethod.GET)
+    @PreAuthorize("hasRole('ADMIN') or (hasAnyRole('USER', 'EXPERT'))")
+    ResponseEntity<List<ShanoirEvent>> findEventsByStudyId(
+            @Parameter(name = "id of the study", required = true) @PathVariable("studyId") Long studyId) throws RestServiceException;
```

Comment:
I thought we said that we were going to use Pageing loading.
There is potentially more than 100k events for a single study in shanoir (not yet, but as you added study id everywhere it will change), we need to load page by page.
So the research will be done as in examination list, field by field.